pkg/controller/jobframework/reconciler.go

```diff
@@ -413,7 +426,11 @@ func (r *JobReconciler) startJob(ctx context.Context, job GenericJob, object cli
 	if err != nil {
 		return err
 	}
-	job.RunWithPodSetsInfo(info)
+	runErr := job.RunWithPodSetsInfo(info)
+	if runErr != nil {
+		// mark the workload as failed
```

Comment:
It seems `finished` rather than failed,  but I would drop this comment from here as it only propagates the error up, so interpretation happens layer up.